src/shorah/shotgun.py

```diff
@@ -589,30 +593,28 @@ def main(args):
     ccx = {}
     # handle case where bamfile has no dots in name
     cin_stem = re.sub(r'\.[^.]+$', r'', os.path.split(in_bam)[1])
-    fch = open('%s.cor.fas' % cin_stem, 'w')
     logging.debug('writing to file %s.cor.fas', cin_stem)
-    for ID, seq_list in to_correct:
-        cor_read = ''.join(seq_list)
-        init_x = len(cor_read.lstrip('-')) - len(cor_read.lstrip('-X'))
-        fin_x = len(cor_read.rstrip('-')) - len(cor_read.rstrip('-X'))
-        cx = seq_list.count('X') - init_x - fin_x
-        ccx[cx] = ccx.get(cx, 0) + 1
-        if cx <= min_x_thresh and cor_read.lstrip('-X') != '':
-            fch.write('>%s %d\n' % (ID, aligned_reads[ID][2] + init_x
-                                    - aligned_reads[ID][0]))
-            cc = 0
-            for c in cor_read.lstrip('-X'):
-                if c != 'X':
-                    fch.write(str(c))
-                    fch.flush()
-                    cc = cc + 1
-                    if cc % fasta_length == 0:
-                        fch.write('\n')
-
-            if cc % fasta_length != 0:
-                fch.write('\n')
-    print(ccx)
-    fch.close()
+    with open('%s.cor.fas' % cin_stem, 'w') as fch:
```

Comment:
Use an f-string here, or at least `str.format`.

Reply:
previous code, prefer not to touch it